src/Select.purs

```diff
@@ -22,124 +25,94 @@ import Effect.Ref (Ref)
 import Effect.Ref as Ref
 import Halogen as H
 import Halogen.HTML as HH
-import Renderless.State (getState, modifyState_, modifyStore)
+import Halogen.Query.ChildQuery (ChildQueryBox)
+import Renderless.State (getState, modifyState_, modifyStore_)
+import Unsafe.Coerce (unsafeCoerce)
 import Web.Event.Event (preventDefault)
-import Web.HTML.HTMLElement (blur, focus)
+import Web.HTML.HTMLElement as HTMLElement
 import Web.UIEvent.KeyboardEvent as KE
 import Web.UIEvent.MouseEvent as ME
 
-----------
--- Component Types
+-----
+-- ACTIONS
 
--- | A useful shorthand for the Halogen component type
-type Component o item m = H.Component HH.HTML (Query o item) (Input o item) (Message o item) m
+newtype Action item v ps m = Action (Variant
```

Comment:
This newtype is necessary because of the self-reference in `andThen`. There is also a mutual reference with `Input` because of `receive` which is a record type synonym, so if this isn't newtyped then at least that will need to be.